CloudNotes/CloudNotes/Utility/CoreDataManager.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+import CoreData
+
+class CoreDataManager {
```

Comment:
DM으로 알려주신 것처럼 `StorageProtocol`을 추가하여 CRUD 기능 (변하지 않는 것)을 정의하고,
`MemoCoreDataManager` 클래스 (변할 수 있는 것)가 해당 프로토콜을 채택하도록 하도록 했습니다.
(원래 `CoreDataManagerProtocol`도 추가하고, 제네럴 메서드로 CoreData CRUD 기능을 기본 구현하려고 했지만, Memo 데이터만 취급하고 있는 현재 프로젝트에서는 무의미할 것 같아서 이 과정은 생략했습니다.)

`MasterTableViewDataSource` 내부에 `memoDataManager` 프로퍼티를 추가하며, 프로퍼티 타입을 `StorageProtocol`로 지정 (이니셜라이저에서 `MemoCoreDataManager` 인스턴스를 기본값으로 할당)해주면 의존성을 낮출 수 있을 것 같아서 시도해봤습니다. 그런데 제네릭 관련 "protocol can only be used as a generic constraint because it has Self or associated type requirements” 에러를 해결할 수 없어서 조언을 구하고 싶습니다. 😭